classes/utility.php

```diff
@@ -217,6 +217,29 @@ public function delete_items($whereparams=null) {
             return;
         }
 
+        // Update completion state: step 1 of 2.
+        // Before deleting items, get the list of involved students as they may get course completion.
+        // After items deletion you will no longer be able to select them with this query.
+        $completion = new completion_info($COURSE);
+        if ($completion->is_enabled($this->cm) && $this->surveypro->completionsubmit) {
+            $sql = 'SELECT DISTINCT s.userid as id
+                    FROM {surveypro_submission} s';
+            if (count($whereparams) == 1) {
```

Comment:
Note I'm not a fan of deciding logic based in the number of params. If for any reason a new mandatory parameter is added, it's really easy to forget that you've code based in the count(). Instead, you should be looking for existing params (`if (isset($whereparams['xxxxx']) then .. else if ... else ....`)

Anyway, I'm not going to reject this because of that, just commenting it's always better to do things as deterministic as possible.